src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -55,215 +64,289 @@ void apply() {
+                }
+                pipelineName.ifPresent(value ->
+                        customValueSearchLinker.addCustomValueAndSearchLink("CI pipeline", value));
+                jobName.ifPresent(value ->
+                        customValueSearchLinker.addCustomValueAndSearchLink("CI job", value));
+                stageName.ifPresent(value ->
+                        customValueSearchLinker.addCustomValueAndSearchLink("CI stage", value));
             }
-            pipelineName.ifPresent(value ->
-                customValueSearchLinker.addCustomValueAndSearchLink("CI pipeline", value));
-            jobName.ifPresent(value ->
-                customValueSearchLinker.addCustomValueAndSearchLink("CI job", value));
-            stageName.ifPresent(value ->
-                customValueSearchLinker.addCustomValueAndSearchLink("CI stage", value));
-        }
 
-        if(CiUtils.isAzurePipelines(providers)) {
-            Optional<String> azureServerUrl = envVariable("SYSTEM_TEAMFOUNDATIONCOLLECTIONURI");
-            Optional<String> azureProject = envVariable("SYSTEM_TEAMPROJECT");
-            Optional<String> buildId = envVariable("BUILD_BUILDID");
-            if (Stream.of(azureServerUrl, azureProject, buildId).allMatch(Optional::isPresent)) {
-                //noinspection OptionalGetWithoutIsPresent
-                String buildUrl = String.format("%s%s/_build/results?buildId=%s",
-                    azureServerUrl.get(), azureProject.get(), buildId.get());
-                buildScan.link("Azure Pipelines build", buildUrl);
-            } else if (azureServerUrl.isPresent()) {
-                buildScan.link("Azure Pipelines", azureServerUrl.get());
+            if(CiUtils.isAzurePipelines(providers)) {
+                Optional<String> azureServerUrl = Utils.envVariable("SYSTEM_TEAMFOUNDATIONCOLLECTIONURI", providers);
+                Optional<String> azureProject = Utils.envVariable("SYSTEM_TEAMPROJECT", providers);
+                Optional<String> buildId = Utils.envVariable("BUILD_BUILDID", providers);
+                if (Stream.of(azureServerUrl, azureProject, buildId).allMatch(Optional::isPresent)) {
+                    //noinspection OptionalGetWithoutIsPresent
+                    String buildUrl = String.format("%s%s/_build/results?buildId=%s",
+                            azureServerUrl.get(), azureProject.get(), buildId.get());
+                    buildScan.link("Azure Pipelines build", buildUrl);
+                } else if (azureServerUrl.isPresent()) {
+                    buildScan.link("Azure Pipelines", azureServerUrl.get());
+                }
+
+                buildId.ifPresent(value ->
+                        buildScan.value("CI build number", value));
             }
+        }
 
-            buildId.ifPresent(value ->
-                buildScan.value("CI build number", value));
+        private static Properties readPropertiesFile(String fileName, ProviderFactory providers, Directory projectDirectory) {
```

Comment:
what does this method declaration do? it seems like just a redirect

Reply:
Absolutely, this was already there before the PR
https://github.com/gradle/common-custom-user-data-gradle-plugin/blob/97e48144a0d4370d3e2bb409bcba02878adb4aa7/src/main/java/com/gradle/CustomBuildScanEnhancements.java#L527

Reply:
Just removed the indirection, that's better now 👍